viewer/packages/api/src/api-helpers/Image360ApiHelper.ts

```diff
@@ -125,20 +132,27 @@ export class Image360ApiHelper {
   }
 
   public async enter360Image(image360Entity: Image360Entity): Promise<void> {
-    const lastEntered360ImageEntity = this._interactionState.currentImage360Entered;
-    this._interactionState.currentImage360Entered = image360Entity;
-
-    if (lastEntered360ImageEntity === image360Entity) {
+    const lastClicked360ImageEntity = this._interactionState.currentImage360Clicked;
+    if (lastClicked360ImageEntity === image360Entity) {
       this._requestRedraw();
       return;
     }
+    this._interactionState.currentImage360Clicked = image360Entity;
 
     await this._image360Facade.preload(image360Entity);
 
-    if (this._interactionState.currentImage360Entered !== image360Entity) {
+    if (this._interactionState.currentImage360Clicked !== image360Entity) {
```

Comment:
Due to the await above (which can take a long time if we are downloading full resolution images) it is very possible to click many icons before this code executes. We only want to enter the last clicked, so that is what this check is for.